packages/nimble-components/src/table/index.ts

```diff
@@ -194,16 +258,44 @@ export class Table<
         }
     }
 
+    /** @internal */
+    public async onRowClick(rowIndex: number): Promise<void> {
+        if (this.selectionMode === TableRowSelectionMode.none) {
+            return;
+        }
+
+        const row = this.table.getRowModel().rows[rowIndex];
+        if (!row) {
+            return;
+        }
+
+        const currentSelection = await this.getSelectedRecordIds();
+        if (currentSelection.length === 1 && currentSelection[0] === row.id) {
+            // The clicked row is already the only selected row. Do nothing.
+            return;
+        }
```

Comment:
Just out of curiosity, the above `if` check is going to be useful for the multi-select case I suppose? Otherwise I would have suspected we would just be able to check the selected state of `row` itself, and then just return if it was selected (thus avoiding the need for this to be async). Not a big deal, just curious if I'm not considering the actual use case that merits us retrieving all of the selected rows in the single-select case.

Reply:
Good point. Since we are in the single-select case here, we could base the selection on the current row and then also pass that selection to `emitSelectionChangeEvent()`. If you don't have a strong preference, I'm tempted to leave this as-is since the implementation will need to look like this once we have multi-select support. Otherwise, I'll have to change a bunch of code to be non-async just to make it all async again in my next PR.

Reply:
No, it's fine. Just wanted to make sure I understood why it was the way it was.